### DISCUSSION TITLE
Give plugins access to conversation history

### DISCUSSION THREAD

**Discussion by someone13574** [LAION-AI/Open-Assistant#3366] [Ideas]
Some plugins could benefit from having access to the full conversation history. This could be especially helpful in plugins aiming to improve long context. To enable this feature, I propose adding an option to the `ai-plugin.json` file that enables plugins to read the full conversation history.

To protect user privacy, we could add a small warning message, informing users that enabling this feature would allow the plugin to access the full history of that particular conversation. The full conversation history would be passed as an additional parameter to each function, stripped from OA's view of the OpenAPI description, and passed automatically when any function in that plugin is called.

This change would provide significant value to plugin developers and enable them to build more context-aware plugins.

**Comment by olliestanley** [comment]
The `ai-plugin.json` file isn't really our spec to mess around with, but afaik we could add chat history anyway if we had the context to do it, we just don't have that at the moment.

(edit: looking back at this I think I misinterpreted what was meant, I guess this would be possible without higher ctx length)
